src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -122,57 +122,66 @@ const StudentApplication = ({ userProfile, authentication }) => {
   const [snackbarSeverity, setSnackbarSeverity] = useState('');
   const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
 
-  // const peopleSearchRef = useRef();
-
   // Off-campus program info is stored as a Map, where the Key is a student's username and the corresponding Value is the department of that student's off-campus program
   let offCampusProgramInfo = new Map();
   applicants.forEach((profile) => offCampusProgramInfo.set(profile.AD_Username, ''));
 
-  /**
-   * Attempt to load an existing application from the database if one exists
-   */
-  const loadSavedApplication = useCallback(async () => {
-    // TODO: Implement this once save/load of application data has been implemented in the backend
-    setLoading(true);
-    // Check if the current user is on an application. Returns the application ID number if found
-    let newApplicationID = -1; // await housing.getApplicationID();
-    if (newApplicationID !== null && newApplicationID !== -1) {
-      setApplicationID(newApplicationID);
-      let applicationDetails = await housing.getApartmentApplication(newApplicationID);
-      if (applicationDetails) {
-        if (applicationDetails.DateSubmitted) {
-          setDateSubmitted(applicationDetails.DateSubmitted);
-        }
-        if (applicationDetails.DateModified) {
-          setDateModified(applicationDetails.DateModified);
+  useEffect(() => {
+    /**
+     * Loads the user's saved apartment application, if one exists
+     */
+    const loadSavedApplication = async () => {
+      // TODO: Implement this once save/load of application data has been implemented in the backend
+      setLoading(true);
+      // Check if the current user is on an application. Returns the application ID number if found
+      let newApplicationID = null;
+      try {
+        newApplicationID = await housing.getApplicationID();
+        console.log('Retrieved Application ID: ' + newApplicationID);
+        if (newApplicationID === null || newApplicationID === -1) {
+          // Intentionally trigger the 'catch'
+          throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
         }
-        if (applicationDetails.Username) {
-          setEditorUsername(applicationDetails.Username);
+        setApplicationID(newApplicationID);
+        let applicationDetails = await housing.getApartmentApplication(newApplicationID);
+        if (applicationDetails) {
+          if (applicationDetails.DateSubmitted) {
+            setDateSubmitted(applicationDetails.DateSubmitted);
+          }
+          if (applicationDetails.DateModified) {
+            setDateModified(applicationDetails.DateModified);
+          }
+          if (applicationDetails.Username) {
+            setEditorUsername(applicationDetails.Username);
+          }
+          if (applicationDetails.Applicants) {
+            setApplicants(applicationDetails.Applicants);
+          }
```

Comment:
Is the way you are doing this if statement the most efficient way of setting everything?

Reply:
This approach is for the sake of resilience/reliability at the expense of some efficiency. This is also needed because it is often possible for only some of the properties of `applicantionDetails` to have real values (Example: if you are loading an application that has applicants saved but no halls saved yet)
The `DateSubmitted` is also expected to be false or null or empty most of the time, so it is best to check the parts of the applicationDetails individually to make sure things always behave the way that they are supposed to